Layers/Connections/PolygonObjectToRepeatingObject.gd

```diff
@@ -0,0 +1,143 @@
+extends LayerCompositionConnection
+class_name PolygonObjectToRepeatingObject
+
+
+var instanced_geo_lines := {}
+
+
+func _build_grid_from_names(node_names: Array) -> Dictionary:
+	var grid = {}
+	
+	for node_name in node_names:
+		var x = node_name.split("_")[0]
+		var y = node_name.split("_")[1]
+		
+		if not x in grid:
+			grid[x] = {}
+		
+		grid[x][y] = true
+	
+	return grid
+
+
+func _count_neighbors(grid: Dictionary) -> Dictionary:
+	var neighbor_counts = {}
+	
+	# Define possible neighbor offsets (direct + diagonal)
+	var offsets = [
+		Vector2(-1, -1), Vector2(0, -1), Vector2(1, -1), # Top row (left, center, right)
+		Vector2(-1, 0),                  Vector2(1, 0),  # Middle row (left, right)
+		Vector2(-1, 1),  Vector2(0, 1),  Vector2(1, 1)   # Bottom row (left, center, right)
+	]
+
+	for x in grid.keys():
+		for y in grid[x].keys():
+			var pos := Vector2(int(x), int(y))
+			var count := 0
+
+			for offset in offsets:
+				var nx = str(int(pos.x + offset.x))
+				var ny = str(int(pos.y + offset.y))
+
+				# Ensure the neighbor exists in the grid
+				if grid.has(nx) and grid[nx].has(ny):
+					count += 1
+
+			# Store neighbor counts
+			if not neighbor_counts.has(x):
+				neighbor_counts[x] = {}
+			
+			neighbor_counts[x][y] = count
+
+	return neighbor_counts
+
+
+func _filter_grid_by_num_neighbors(grid, neighbor_counts, num_neighbors: int):
+	for x in neighbor_counts.keys():
+		for y in neighbor_counts[x].keys():
+			if neighbor_counts[x][y] <= num_neighbors:
+				grid[x].erase(y)
+
+
+func extract_relevant_data(source: LayerComposition, 
+							new_features: Array, 
+							removed_features: Array) -> Variant:
+	# Filter new features for a set "outline" field to create a barrier around it
+	var relevant_new = new_features.filter(func(feature: GeoFeature): 
+				return feature.get_attribute("outline") != "")
+	var relevant_old = removed_features.filter(func(feature: GeoFeature): 
+				return feature.get_attribute("outline") != "")
+	return {"new": relevant_new, "removed": relevant_old}
+
+
+func apply_to_target(target: LayerComposition, features: Variant):
+	var new_features = features["new"]
+	var removed_features = features["removed"]
+	
+	# Remove all outdated features
+	for feature in removed_features:
+		var line_layer: GeoFeatureLayer = target.render_info.geo_feature_layer
+		var geo_line = instanced_geo_lines[feature.get_id()]
```

Comment:
It feels like it might be possible that the PolygonObjectRenderer instances and deletes a feature before this Connection has had an opportunity to handle it, therefore causing the `instanced_geo_lines` array indexing to fail. But I can't think of a case where this would actually be possible since we always directly react to apply_new_data, so maybe it's fine.